smartsim/experiment.py

```diff
@@ -160,28 +174,50 @@ def __init__(
 
         self.exp_path = exp_path
 
-        self._launcher = launcher.lower()
-
-        if self._launcher == "auto":
-            self._launcher = detect_launcher()
-        if self._launcher == "cobalt":
-            raise SSUnsupportedError("Cobalt launcher is no longer supported.")
+        # TODO: Remove this! The contoller is becoming obsolete
+        self._control = Controller(launcher="local")
+        self._dispatcher = settings_dispatcher
 
-        if launcher == "dragon":
-            self._set_dragon_server_path()
-
-        self._control = Controller(launcher=self._launcher)
+        self._active_launchers: set[LauncherLike[t.Any]] = set()
+        """The active launchers created, used, and reused by the experiment"""
 
         self.fs_identifiers: t.Set[str] = set()
         self._telemetry_cfg = ExperimentTelemetryConfiguration()
 
-    def _set_dragon_server_path(self) -> None:
-        """Set path for dragon server through environment varialbes"""
-        if not "SMARTSIM_DRAGON_SERVER_PATH" in environ:
-            environ["SMARTSIM_DRAGON_SERVER_PATH_EXP"] = osp.join(
-                self.exp_path, CONFIG.dragon_default_subdir
+    def start_jobs(self, *jobs: Job) -> tuple[LaunchedJobID, ...]:
+        """WIP: replacemnt method to launch jobs using the new API"""
+
+        if not jobs:
+            raise TypeError(
+                f"{type(self).__name__}.start_jobs() missing at least 1 required "
+                "positional argument"
             )
 
+        def _start(job: Job) -> LaunchedJobID:
+            builder: LaunchArgBuilder[t.Any] = job.launch_settings.launch_args
+            launcher_type = self._dispatcher.get_launcher_for(builder)
+            launcher = first(
+                lambda launcher: type(launcher) is launcher_type,
+                self._active_launchers,
+            )
+            if launcher is None:
```

Comment:
Per offline discussion, decided to keep passing in the full `Experiment` so that new launchers could access other public attrs other than just the experiment name/path (should we choose to add them in future).

Resolving for now, but feel free to re-open if we think there is more to discuss!